### PR TITLE
datadog: support the agent6 in the helm chart

### PR DESCRIPTION
**Pull request by JulienBalestra**
### What does this PR do?

This PR changes the datadog chart to run the datadog/agent in Kubernetes.

### CODE REVIEW

**Comment by unguiculus** [comment]
cc @hkaj @irabinovitch @xvello

I noticed the chart uses a `latest` image. This is not ideal. Please consider changing this to a release version.

File: stable/datadog/templates/daemonset.yaml [hunk 1]
```diff
@@ -69,16 +73,29 @@ spec:
...
+          {{- if .Values.datadog.collectEvents }}
+          - name: DD_COLLECT_KUBERNETES_EVENTS # agent6
+            value: {{.Values.datadog.collectEvents | quote}}
           {{- end }}
-          - name: SD_BACKEND
+          - name: SD_BACKEND  # agent5
             value: docker
           - name: KUBERNETES
             value: "yes"
+          - name: DD_KUBERNETES_KUBELET_HOST
```
Comment (xvello) [inline stable/datadog/templates/daemonset.yaml]: Can you guard this behind a 1.7+ version check?
See https://github.com/kubernetes/charts/blob/master/stable/datadog/templates/_helpers.tpl#L46 for how to do it

File: stable/datadog/templates/daemonset.yaml [hunk 2]
```diff
@@ -58,7 +58,11 @@ spec:
             value: {{ .Values.datadog.logLevel | quote }}
           {{- end }}
           {{- if .Values.datadog.nonLocalTraffic }}
-          - name: NON_LOCAL_TRAFFIC
+          - name: NON_LOCAL_TRAFFIC # agent5
+            value: {{ .Values.datadog.nonLocalTraffic | quote }}
+          {{- end }}
+          {{- if .Values.datadog.nonLocalTraffic }}
```
Comment (xvello) [inline stable/datadog/templates/daemonset.yaml]: I think these two `{{- if .Values.datadog.nonLocalTraffic }}` can be merged

**Review by xvello — COMMENTED** [review]
As clean as we can do while keeping agent5 compat 👍 
Two small changes

File: stable/datadog/templates/daemonset.yaml (hunk 1 above)
Comment (unguiculus) [reply stable/datadog/templates/daemonset.yaml]: This won't work anymore once 1.10 is out. You need to use `semvercompare`. See https://github.com/kubernetes/charts/pull/3369/files#diff-d0b2d71f2c76df54f944e5d7204e9f94R44.

**Comment by xvello** [comment]
Indeed: https://github.com/kubernetes/helm/blob/master/docs/chart_best_practices/pods.md#images

Let's put 6.0.0 and document `latest` can be used to. Let's also quickly document the `-jmx` variant

**Comment by so0k** [comment]
Doesn't leader election also require a configmap named `datadog-leader-election` to be created?

https://docs.datadoghq.com/agent/basic_agent_usage/kubernetes/#event-collection

> A ConfigMap can be used to store the `event.tokenKey` and the `event.tokenTimestamp`. It has to be deployed in the default namespace and be named `datadog-leader-election`. One can simply run
>```bash
>kubectl create configmap datadogtoken --from-literal="event.tokenKey"="0"
>```
>You can also use the example in manifests/datadog_configmap.yaml.

See [leader election](https://sourcegraph.com/github.com/DataDog/datadog-agent@6.1.4/-/blob/pkg/util/kubernetes/apiserver/leaderelection/leaderelection.go#L32)

**Comment by xvello** [comment]
@so0k That configmap is automatically created if not found, see https://github.com/DataDog/datadog-agent/blob/b65371ebad6b0ad80fd6f2fd8ae76bbd4f37a80d/pkg/util/kubernetes/apiserver/leaderelection/leaderelection_engine.go#L52

This is why we grand [configmap create permission](https://github.com/kubernetes/charts/blob/6dceaa196ab9c5f531371f3d05d1d60dbfbb1c02/stable/datadog/templates/clusterrole.yaml#L48-L52) in the ClusterRole. The documentation has been updated to fix the configmap name. Sorry for the confusion.

**Comment by so0k** [comment]
I must have missed that, thanks for the update

PS: I guess this code is from before announcements of metaparticle and more generic leaderElection libraries provided by k8s? (or perhaps it is done through a configmap as you wish to keep track of the last events parsed?)

**Pull request merged**